var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -69,6 +69,8 @@ class Cmake(Package):
     # Cannot build with Intel, should be fixed in 3.6.2
     # https://gitlab.kitware.com/cmake/cmake/issues/16226
     patch('intel-c-gnu11.patch', when='@3.6.0:3.6.1')
+    patch('ibm-xl.patch', when='%xl')
```

Comment:
looks like the patches are only for `3.5.2`, are they? if so, you probably want to add 
```
when=`@3.5.2%xl`
```